lib/validations.rb

```diff
@@ -1,10 +1,89 @@
 class Validations
 
   def validate_ship_number(number)
-    number = number.to_i
-    valid_numbers = [1, 2, 3, 4]
-    if !valid_numbers.include?(number)
-      :error
+    valid_options = [1, 2, 3, 4]
+    if !valid_options.include?(number.to_i)
+      :invalid_ship_number
+    else
+      :valid_ship_number
+    end
+  end
+
+  def validate_position_for_ship(ship, input)
+    if validate_position(input) == :invalid_ship_position
+      :invalid_ship_position
+    elsif check_ship_is_inside_grid(ship, input) == :invalid_placement
+      :invalid_placement
+    else
+      :valid_ship_position
+    end
+  end
+
+  def validate_position_to_attack(input)
+    if !valid_number?(input[0]) || !valid_letter?(input[1])
+      :invalid_attack
+    else
+      :valid_attack
+    end
+  end
+
+  private
+
+  def not_inside_grid?(orientation, ship_length, letter, number)
+    orientation == :horizontal ? invalid_horizontally?(ship_length, number) : invalid_vertically?(ship_length, letter)
+  end
+
+  def invalid_vertically?(ship_length, letter)
+    (number_for[letter.downcase] + (ship_length-1)) > 10
+  end
+
+  def invalid_horizontally?(ship_length, number)
+    (number.to_i + ship_length-1) > 10
+  end
+
+  def number_for
```

Comment:
This is duplicated from `GridDisplay`. Can you use the method from `GridDisplay` instead? The only difference is that you're using lower case letters here. 
It's not good to duplicate code because that means, if the grid changes, let's say we make it bigger... then you have to remember to update this here as well. So maybe better to pass in the `GridDisplay` here?